docs/tutorials/services/source-ip.md

```diff
@@ -0,0 +1,338 @@
+
+
+### Prerequisites
+
+You must have a working Kubernetes 1.5 cluster to run the examples in this
+document. The examples use a small nginx webserver that echoes back the source
+IP of requests it receives through a HTTP header. You can create it as follows:
+
+```console
+$ kubectl run source-ip-app --image=gcr.io/google_containers/echoserver:1.4
+deployment "source-ip-app" created
+```
+
+{% endcapture %}
+
+{% capture objectives %}
+
+* Expose a simple application through various types of Services
+* Understand how each Service type handles source IP NAT
+* Understand the tradeoffs involved in preserving source IP
+
+{% endcapture %}
+
+
+{% capture lessoncontent %}
+
+### Source IP for Services with Type=ClusterIP
+
+Packets sent to the ClusterIP of a Service are never source NAT'd if you're
+running kube-proxy in [iptables mode](/docs/user-guide/services/#proxy-mode-iptables). Kube-proxy
+exposes its mode through a `proxyMode` endpoint:
+
+```console
+$ kubectl get no
+NAME                           STATUS                     AGE
+kubernetes-minion-group-6jst   Ready                      2h
+kubernetes-minion-group-cx31   Ready                      2h
+kubernetes-minion-group-jj1t   Ready                      2h
+
+kubernetes-minion-group-6jst $ curl localhost:10249/proxyMode
+iptables
+```
+
+You can test source IP preservation by creating a Service over the source IP app:
+
+```console
+$ kubectl expose deployment source-ip-app --name=clusterip --port=80 --target-port=8080
+service "clusterip" exposed
+
+$ kubectl get svc clusterip
+NAME         CLUSTER-IP    EXTERNAL-IP   PORT(S)   AGE
+clusterip    10.0.170.92   <none>        80/TCP    51s
+```
+
+And hitting the `ClusterIP` from a pod in the same cluster
+
+```console
+$ kubectl run busybox -it --image=busybox --restart=Never --rm
+Waiting for pod default/busybox to be running, status is Pending, pod ready: false
+If you don't see a command prompt, try pressing enter.
+
+# ip addr
+1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue
+    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
+    inet 127.0.0.1/8 scope host lo
+       valid_lft forever preferred_lft forever
+    inet6 ::1/128 scope host
+       valid_lft forever preferred_lft forever
+3: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1460 qdisc noqueue
+    link/ether 0a:58:0a:f4:03:08 brd ff:ff:ff:ff:ff:ff
+    inet 10.244.3.8/24 scope global eth0
+       valid_lft forever preferred_lft forever
+    inet6 fe80::188a:84ff:feb0:26a5/64 scope link
+       valid_lft forever preferred_lft forever
+
+# wget -qO - 10.0.170.92
+CLIENT VALUES:
+client_address=10.244.3.8
+command=GET
+...
+```
+
+### Source IP for Services with Type=NodePort
+
+As of Kubernetes 1.5, packets sent to Services with [Type=NodePort](/docs/user-guide/services/#type-nodeport)
+are source NAT'd by default. You can test this by creating a `NodePort`
+Service
```

Comment:
Hanging cr/lf after `NodePort`; please delete and add a colon (:) after "Service".